clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestFuture.java

```diff
@@ -100,11 +100,10 @@ public RuntimeException exception() {
      * @param value corresponding value (or null if there is none)
      */
     public void complete(T value) {
-        if (isDone)
+        if (!isDone.compareAndSet(false, true))
```

Comment:
why do we use `isDone` here instead of having an `AtomicReference` to `value`? aren't we in a state where we `isDone()` returns `true` and `value()` returns null between lines 102/104